collectd/files/ping.conf

```diff
@@ -1,17 +1,30 @@
+{%- from "collectd/map.jinja" import collectd_settings with context %}
+{%- set ping_settings = collectd_settings.plugins.ping %}
+
+{%- set hfg = ping_settings.hosts_from_grains %}
+
 {% if hfg and salt['mine.get'](hfg.target, hfg.fun, hfg.expr_form) -%}
 {% set hosts = hosts + salt['mine.get'](hfg.target, hfg.fun, hfg.expr_form).values() -%}
 {% endif -%}
+
+#
+# DO NOT EDIT
+#
+# This file is managed by salt via {{ source }}
+# Modify the config that generates this file instead
+#
+
 LoadPlugin ping
+
 <Plugin "ping">
-  {%- for host in hosts %}
+  {%- for host in ping_settings.hosts %}
   Host "{{ host }}"
```

Comment:
The stuff above (hosts_from_grains) adds some hosts to what was the hosts pillar setting. So you either need to add the mine.get stuff to ping_settings.hosts or rework how you're fetching hosts.